### PR TITLE
[dapp-kit-next] Unit & Integration tests groundwork

### PR DESCRIPTION
**Pull request by gwro**
## Description

Describe the changes or additions included in this PR.

## Test plan

How did you test the new or updated feature?

---

### CODE REVIEW

File: packages/dapp-kit-next/packages/dapp-kit-core/test/integration/setup.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { afterEach, beforeEach } from 'vitest';
+
+export function suppressFetchErrorsDuringTeardown() {
```
Comment (hayes-mysten) [inline packages/dapp-kit-next/packages/dapp-kit-core/test/integration/setup.ts:6]: what fetch requests are we actually making, is this for slush web?  It seems like we should be able to test/use most of dapp-kit without it making a lot of requests (I might be completely wrong here, its been a while since I looked at dapp-kit stuff)

File: packages/dapp-kit-next/packages/dapp-kit-core/test/unit/stores/compatible-wallets.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import { createMockAccount } from '../../mocks/mock-account';
+import type { MockWalletOptions } from '../../mocks/mock-wallet';
+import {
+	StandardConnect,
+	SuiSignTransaction,
+	SuiSignTransactionBlock,
+} from '@mysten/wallet-standard';
+import { excludeUiWalletsByName } from '../../test-utils';
+
+describe('[Unit] $compatibleWallets', () => {
```
Comment (williamrobertson13) [inline packages/dapp-kit-next/packages/dapp-kit-core/test/unit/stores/compatible-wallets.test.ts:15]: I think these tests are useful 😄, I do wonder if it would be easier / better value to just test these as part of creating the whole dApp Kit instance but some of these cases are more specific edge cases so I'm fine with the setup here!

File: packages/dapp-kit-next/packages/dapp-kit-core/test/mocks/mock-ui-wallet.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+	return {
+		address,
+		publicKey,
+		chains,
+		features,
+	} as UiWallet['accounts'][number];
+}
+
+export function createMockUiWallet(options: MockUiWalletOptions = {}): UiWallet {
```
[the remark this answers is not in the document]
Comment (gwro) [reply packages/dapp-kit-next/packages/dapp-kit-core/test/mocks/mock-ui-wallet.ts:40]: Sorry, it's just some leftovers from when I was learning the code in the beginning 😁 Removing 💯
